### PR TITLE
add new string words

### PR DESCRIPTION
**Pull request by cryshado**
- `$at (S n -- S')`
- `$mul (S n -- S*n)`
- `$sybs (S -- t[S'0, S'1, S'2, ..., S'n])`
- `$sub (S x y -- S')`
- `$sep (S S1 -- t(...))`
- `$rep  (S S1 S2   -- S')`
- `$repn (S S1 S2 n -- S')`
- `$repm (S S1 S2 -- S')`

### CODE REVIEW

File: src/modules/string_utils.rs [hunk 1]
```diff
@@ -162,6 +162,84 @@ impl StringUtils {
...
 
+    // $at (S n -- S')
+    #[cmd(name = "$at", stack)]
+    fn interpret_str_at(stack: &mut Stack) -> Result<()> {
+        let index = stack.pop_usize()?;
+        let string = stack.pop_string()?;
+
+        match string.chars().nth(index) {
+            Some(s) => stack.push(s.to_string()),
+            None => anyhow::bail!("index must be >= 0 and <= {}", usize::MAX),
```
Comment (Rexagon) [inline src/modules/string_utils.rs]: `index` range is guaranteed to be `0..usize::MAX`, it is better to change error to `Index out of string bounds`

File: src/modules/string_utils.rs [hunk 2]
```diff
@@ -162,6 +162,84 @@ impl StringUtils {
...
+    }
+
+    // $sub (S x y -- S')
+    #[cmd(name = "$sub", stack)]
+    fn interpret_str_sub(stack: &mut Stack) -> Result<()> {
+        let y = stack.pop_usize()?;
+        let x = stack.pop_usize()?;
+        let string = stack.pop_string()?;
+
+        stack.push(string[x..y].to_string())
```
Comment (Rexagon) [inline src/modules/string_utils.rs:213]: Please add a bounds check here as it will panic by default

File: src/modules/string_utils.rs (hunk 1 above)
Comment (cryshado) [reply src/modules/string_utils.rs]: FIXED

File: src/modules/string_utils.rs (hunk 2 above)
Comment (cryshado) [reply src/modules/string_utils.rs:213]: FIXED

**Pull request merged**